crates/primitives/src/transaction/pooled.rs

```diff
@@ -527,46 +485,39 @@ impl Decodable for PooledTransactionsElement {
                         hash: typed_tx.hash,
                     }),
                     #[cfg(feature = "optimism")]
-                    Transaction::Deposit(tx) => Ok(PooledTransactionsElement::Deposit {
-                        transaction: tx,
-                        signature: typed_tx.signature,
-                        hash: typed_tx.hash,
-                    }),
+                    Transaction::Deposit(_) => Err(RlpError::Custom("Optimism deposit transaction cannot be decoded to PooledTransactionsElement"))
                 }
             }
         }
     }
 }
 
-impl From<TransactionSigned> for PooledTransactionsElement {
-    /// Converts from a [TransactionSigned] to a [PooledTransactionsElement].
-    ///
-    /// NOTE: For EIP-4844 transactions, this will return an empty sidecar.
-    fn from(tx: TransactionSigned) -> Self {
+impl TryFrom<TransactionSigned> for PooledTransactionsElement {
+    type Error = TransactionConversionError;
+
+    fn try_from(tx: TransactionSigned) -> Result<Self, Self::Error> {
         let TransactionSigned { transaction, signature, hash } = tx;
         match transaction {
             Transaction::Legacy(tx) => {
-                PooledTransactionsElement::Legacy { transaction: tx, signature, hash }
+                Ok(PooledTransactionsElement::Legacy { transaction: tx, signature, hash })
             }
             Transaction::Eip2930(tx) => {
-                PooledTransactionsElement::Eip2930 { transaction: tx, signature, hash }
+                Ok(PooledTransactionsElement::Eip2930 { transaction: tx, signature, hash })
             }
             Transaction::Eip1559(tx) => {
-                PooledTransactionsElement::Eip1559 { transaction: tx, signature, hash }
+                Ok(PooledTransactionsElement::Eip1559 { transaction: tx, signature, hash })
             }
             Transaction::Eip4844(tx) => {
-                PooledTransactionsElement::BlobTransaction(BlobTransaction {
+                Ok(PooledTransactionsElement::BlobTransaction(BlobTransaction {
                     transaction: tx,
                     signature,
                     hash,
-                    // This is empty - just for the conversion!
+                    // Assuming BlobTransaction has a sidecar that needs to be defaulted
                     sidecar: Default::default(),
-                })
+                }))
```

Comment:
this should also return an error now, because we want to prevent this conversion

Reply:
why err? blob transaction can be in request-response?

Reply:
actually bit confused on the reasoning that "deposit tx can't be in p2p so remove from `PooledTransactionsElement`,  is it coz deposit tx will never serve as response for `GetPooledTransactions` since it doesn't need to be propagated at all since it's a layer2 tx so rm from there? but for blob i think it could serve as response for `GetPooledTransactions` just a node shouldn't proactively broadcast it?

Reply:
> coz deposit tx will never serve as response for GetPooledTransactions since it doesn't need to be sent over network at all so rm from there?

correct

> but for blob i think it could serve as response for GetPooledTransactions just a node shouldn't proactively broadcast it?

exactly, but this conversion is a massive footgun for blobtransactions because it sets an empty blob